src/screens/Resources/LoadScreen.js

```diff
@@ -0,0 +1,46 @@
+import firestore from '@react-native-firebase/firestore';
+import { useState, useCallback } from 'react';
+import { useFocusEffect } from 'react-navigation-hooks';
+import PropTypes from 'prop-types';
+import { collections, nav } from '../../constants';
+
+export default function useLoadScreen(navigation) {
+  const [loading, setLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState(null);
+
+  async function firestoreQuery(collection, id) {
+    return firestore()
+      .collection(collection)
+      .where('parent', '==', id)
+      .get();
+  }
+
+  async function loadScreen(id, title) {
```

Comment:
to my knowledge not really. the reason why we handled it here is because we don't know whether we should be routing to a subcategories screen or a resources item list screen. because of that we need to query firebase to see which one we need to go to. and since we query firebase, i took the liberty of just extracting the snapshot and passing it to the next screen anyways. i'm open to any suggestions for how to change this tho!

Reply:
I think we could have a staging screen whose only purpose is to indicate that its in the process of loading the subcategories/resource item list, like how the facebook app shows grey boxes before all the content loads, since I think the issue is mainly with having the activity indicator & error message on the previous screen.

Reply:
hmm i see and then display the error msg and the activity indicator in this staging screen?

Reply:
Yup, exactly